homeassistant/components/sensor/huawei_lte.py

```diff
@@ -0,0 +1,169 @@
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+from ..huawei_lte import DATA_KEY, RouterData
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['huawei_lte']
+
+DEFAULT_NAME_TEMPLATE = 'Huawei {}: {}'
+
+DEFAULT_SENSORS = [
+    "device_information.WanIPAddress",
+    "device_signal.rssi",
+]
+
+SENSOR_META = {
+    "device_information.SoftwareVersion": dict(
+        name="Software version",
+    ),
+    "device_information.WanIPAddress": dict(
+        name="WAN IP address",
+        icon="mdi:ip",
+    ),
+    "device_information.WanIPv6Address": dict(
+        name="WAN IPv6 address",
+        icon="mdi:ip",
+    ),
+    "device_signal.rsrq": dict(
+        name="RSRQ",
+        # http://www.lte-anbieter.info/technik/rsrq.php
+        icon=lambda x:
+        x >= -5 and "mdi:signal-cellular-3"
+        or x >= -8 and "mdi:signal-cellular-2"
+        or x >= -11 and "mdi:signal-cellular-1"
+        or "mdi:signal-cellular-outline"
+    ),
+    "device_signal.rsrp": dict(
+        name="RSRP",
+        # http://www.lte-anbieter.info/technik/rsrp.php
+        icon=lambda x:
+        x >= -80 and "mdi:signal-cellular-3"
+        or x >= -95 and "mdi:signal-cellular-2"
+        or x >= -110 and "mdi:signal-cellular-1"
+        or "mdi:signal-cellular-outline"
+    ),
+    "device_signal.rssi": dict(
+        name="RSSI",
+        # https://eyesaas.com/wi-fi-signal-strength/
+        icon=lambda x:
+        x >= -60 and "mdi:signal-cellular-3"
+        or x >= -70 and "mdi:signal-cellular-2"
+        or x >= -80 and "mdi:signal-cellular-1"
+        or "mdi:signal-cellular-outline"
+    ),
+    "device_signal.sinr": dict(
+        name="SINR",
+        # http://www.lte-anbieter.info/technik/sinr.php
+        icon=lambda x:
+        x >= 10 and "mdi:signal-cellular-3"
+        or x >= 5 and "mdi:signal-cellular-2"
+        or x >= 0 and "mdi:signal-cellular-1"
+        or "mdi:signal-cellular-outline"
+    ),
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_URL): cv.url,
+    vol.Optional(
+        CONF_MONITORED_CONDITIONS, default=DEFAULT_SENSORS): cv.ensure_list,
+})
+
+
+def setup_platform(
+        hass, config, add_entities, discovery_info):
+    """Set up Huawei LTE sensor devices."""
+    data = hass.data[DATA_KEY].get_data(config)
+    sensors = []
+    for path in config.get(CONF_MONITORED_CONDITIONS):
+        sensors.append(HuaweiLteSensor(
+            data, path, SENSOR_META.get(path, {})))
+    add_entities(sensors, True)
+
+
+@attr.s
+class HuaweiLteSensor(Entity):
+    """Huawei LTE sensor entity."""
+
+    data = attr.ib(type=RouterData)
+    path = attr.ib(type=list)
+    meta = attr.ib(type=dict)
+
+    _state = attr.ib(init=False, default=STATE_UNKNOWN)
+    _unit = attr.ib(init=False, type=str)
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique ID for sensor."""
+        return "%s_%s" % (
```

Comment:
Please use new style string formatting.

Reply:
PR #16626. However, if new style is preferred, I think it should be documented in https://developers.home-assistant.io/docs/en/development_guidelines.html (and updated to f-strings when we can depend on 3.6+ only :)): I personally happen to have a slight preference towards the old style, but followed this suggestion based on some git greps and seeing that new style formatting appears to be more common.

Reply:
https://github.com/home-assistant/developers.home-assistant/pull/87